### PR TITLE
cpp: update related to v0.2.1

### PR DESCRIPTION
**Pull request by dmehala**
### What does this PR do? What is the motivation?
A new version of `dd-trace-cpp` has been released. This PR includes updates introduced in the new version, such as support for Windows.

### Merge instructions
- [x] Please merge after reviewing

### Additional notes
🪟

### CODE REVIEW

File: content/en/tracing/trace_collection/automatic_instrumentation/dd_libraries/cpp.md [hunk 1]
```diff
@@ -36,31 +36,9 @@ Before you begin, make sure you have already [installed and configured the Agent
...
-  dd-trace-cpp
-  GIT_REPOSITORY https://github.com/DataDog/dd-trace-cpp
-  GIT_TAG        v0.2.0
-  GIT_SHALLOW    ON
-  GIT_PROGRESS   ON
-)
-
-FetchContent_MakeAvailable(dd-trace-cpp)
+Here is an example application that can be used for testing `dd-trace-cpp`.
+This application creates a tracer instance with the default settings and generates a trace with two spans, which will be reported under the service name `my-service`.
```
Comment (maycmlee) [inline content/en/tracing/trace_collection/automatic_instrumentation/dd_libraries/cpp.md:40]:
```suggestion
This application creates a tracer instance with the default settings and generates a trace with two spans, which is reported under the service name `my-service`.
```

File: content/en/tracing/trace_collection/automatic_instrumentation/dd_libraries/cpp.md [hunk 2]
```diff
@@ -138,44 +174,11 @@ cmake -B build .
...
-    auto span_a = tracer.create_span(options);
-    auto span_b = span_a.create_child();
-    span_b.set_name("B");
-    span_b.set_tag("tag", "value");
-  }
-
-  return 0;
-}
-```
+By default `cmake --install` places the shared library and public headers into the appropriate system directories (e.g., `/usr/local/[...]`).
```
Comment (maycmlee) [inline content/en/tracing/trace_collection/automatic_instrumentation/dd_libraries/cpp.md:178]:
```suggestion
By default, `cmake --install` places the shared library and public headers into the appropriate system directories (for example, `/usr/local/[...]`).
```

**Review by maycmlee — COMMENTED** [review]
Just a couple of small suggestions!

**Comment by dmehala** [comment]
@maycmlee good to merge. Thank you!

**Pull request merged**